### PR TITLE
Introduce a CI job to automatically update submodule main branch

### PR DESCRIPTION
**Pull request by lava**
On a push to main, this will check if the commit that the tenzir-plugins submodule points to is the current master commit of the tenzir/tenzir-plugins repository. If not, it will automatically merge the corresponding plugins PR, update the nix settings, and push a new commit to main.

### CODE REVIEW

File: .github/workflows/tenzir.yaml [hunk 1]
```diff
@@ -83,10 +83,60 @@ jobs:
...
+            gh pr --repo tenzir/tenzir-plugins merge $pr_number --merge --match-head-commit ${plugins_sha}
+
+            # Create a new commit for the updated submodule.
+            git checkout main # TODO: Is this necessary or are we already on main?
+            git -C contrib/tenzir-plugins checkout main
+            new_plugins_main_sha=$(git -C contrib/tenzir-plugins rev-parse --short origin/main)
+            nix/update-plugins.sh
+            git add contrib/tenzir-plugins
+            git add nix/
+            git commit -m "Automatic submodule bump from $plugins_sha_short to $new_plugins_main_sha"
```
Comment (mavam) [inline .github/workflows/tenzir.yaml:137]:
```suggestion
            git commit -m "Bump submodule from $plugins_sha_short to $new_plugins_main_sha"
```

Just to keep our git commit conventions in check, i.e., use verbs to denote the primary action.

File: .github/workflows/tenzir.yaml [hunk 2]
```diff
@@ -83,10 +83,60 @@ jobs:
...
+          if [ "$plugins_sha" != "$plugins_main_sha" ]; then
+            # Auto-merge the plugins PR and update 'tenzir-plugins/main'.
+            # gh pr --repo tenzir/tenzir-plugins list
+            pr_number=$(gh pr --repo tenzir/tenzir-plugins list --search $plugins_sha --json number -q .[0].number)
+            echo $pr_number
+            [ $pr_number ] # Assert the pr number was non-empty
+            gh pr --repo tenzir/tenzir-plugins merge $pr_number --merge --match-head-commit ${plugins_sha}
+
+            # Create a new commit for the updated submodule.
+            git checkout main # TODO: Is this necessary or are we already on main?
```
Comment (mavam) [inline .github/workflows/tenzir.yaml:131]: Does this need to be sorted out before merging?

File: .github/workflows/tenzir.yaml [hunk 3]
```diff
@@ -83,10 +83,60 @@ jobs:
...
+          persist-credentials: false
+      - name: Generate a token
+        if: github.event_name == 'push' && github.ref == 'refs/heads/main'
+        id: generate_token
+        uses: actions/create-github-app-token@v1
+        with:
+          app-id: ${{ vars.TENZIR_TESTAPP_APP_ID }}
+          private-key: ${{ secrets.TENZIR_TESTAPP_PRIVATE_KEY }}
+          owner: ${{ github.repository_owner }}
+      - name: Test submodule update
```
Comment (mavam) [inline .github/workflows/tenzir.yaml:104]: The name seems to be very innocuous compared what it actually does. Maybe make it a bit clearer?

File: .github/workflows/tenzir.yaml (hunk 2 above)
Comment (lava) [reply .github/workflows/tenzir.yaml:131]: We can only really test if after it runs on main. But in the worst case it is going to be a no-op, so it shouldn't matter too much either way.

File: .github/workflows/tenzir.yaml [hunk 4]
```diff
@@ -83,10 +83,60 @@ jobs:
...
+          persist-credentials: false
+      - name: Generate a token
+        if: github.event_name == 'push' && github.ref == 'refs/heads/main'
+        id: generate_token
+        uses: actions/create-github-app-token@v1
+        with:
+          app-id: ${{ vars.TENZIR_TESTAPP_APP_ID }}
+          private-key: ${{ secrets.TENZIR_TESTAPP_PRIVATE_KEY }}
+          owner: ${{ github.repository_owner }}
+      - name: Bump plugins submodule
```
Comment (tobim) [inline .github/workflows/tenzir.yaml:104]: Why is all of this implemented as extra steps of the `configure` job?
May I politely ask to move this logic into its own dedicated workflow?

**Review by tobim — CHANGES_REQUESTED** [review]

**Review by dominiklohmann — APPROVED** [review]

**Pull request merged**